.changeset/giant-areas-press.md

```diff
@@ -0,0 +1,18 @@
+---
+'astro': major
+---
+
+Updates endpoints with a file extension to not be accessed with a trailing slash
```

Comment:
```suggestion
Updates trailing slash behavior of endpoint URLs.
```
This is generic enough (it does update the behavior of some URLS) if you want, given the details below! Does that feel specific *enough* to go with the explanation below?